nuheat/thermostat.py

```diff
@@ -174,41 +178,119 @@ def schedule_mode(self, mode):
 
         self.set_data({"ScheduleMode": mode})
 
+    @property
+    def hold_time(self):
+        """
+        Return a datetime for the current temporary hold time (or None)
+        """
+        if self._schedule_mode == config.SCHEDULE_TEMPORARY_HOLD:
+            return self._hold_time
+        else:
+            return None
+
+    @hold_time.setter
+    def hold_time(self, hold):
+        """
+        Set a temporary hold time (and change schedule_mode to Temporary Hold).
+
+        :param hold: datetime for temporary hold_time.
+        """
+        hold_gmt = hold.astimezone(timezone(timedelta(0), 'GMT'))
```

Comment:
🤔 i've not seen this syntax to convert to UTC before. it may be better to just add `pytz` to `requirements` and then use that here. we should be able to do something like `hold.astimezone(pytz.utc)`.

another thing that might work without any additional libraries in python 3.x might be: `hold.astimezone(datetime.timezone.utc)`. have you tried this?

Reply:
I think I did try using datetime.timezone.utc, but then %Z rendered as "UTC" instead of "GMT", IIRC.  Of course, %Z could just be replaced by a literal "GMT" and problem solved.

I think pytz may offer a more rich / accurate representation of timezones.  For example, I think you can create a timezone object with pytz based on a real politically-defined timezone definition (e.g. "US/Central") and you'd get the benefits of accurate daylight-savings time adjustments, etc.  Unfortunately, I don't think we have the information available to the python-nuheat library in order to identify the specific timezone (aside from the time-offset definition, e.g. -06:00, that datetime.timezone appears to handle well enough).

Anyway, let me know what you prefer, and I can change it if needed.  There might be implications for Python 2.7 support, as well (see my response to your comment below).

Reply:
cool, i always use `UTC` over `GMT`... but if these thermostats want `GMT`, then let's keep your code!

Reply:
it might be a good idea to add a comment that informs devs that nuheat expects `GMT` instead of `UTC` so that we don't accidentally break this later.